extensions/extension-editing/src/extensionLinter.ts

```diff
@@ -155,19 +156,33 @@ export class ExtensionLinter {
 				if (activationEventsNode?.type === 'array' && activationEventsNode.children) {
 					for (const activationEventNode of activationEventsNode.children) {
 						const activationEvent = getNodeValue(activationEventNode);
+						// Redundant Implicit Activation
 						if (info.implicitActivationEvents?.has(activationEvent) && redundantImplicitActivationEventPrefixes.some((prefix) => activationEvent.startsWith(prefix))) {
 							const start = document.positionAt(activationEventNode.offset);
 							const end = document.positionAt(activationEventNode.offset + activationEventNode.length);
 							diagnostics.push(new Diagnostic(new Range(start, end), redundantImplicitActivationEvent, DiagnosticSeverity.Warning));
-						} else {
-							for (const implicitActivationEventPrefix of reservedImplicitActivationEventPrefixes) {
-								if (activationEvent.startsWith(implicitActivationEventPrefix)) {
-									const start = document.positionAt(activationEventNode.offset);
-									const end = document.positionAt(activationEventNode.offset + activationEventNode.length);
-									diagnostics.push(new Diagnostic(new Range(start, end), implicitActivationEvent, DiagnosticSeverity.Error));
-								}
+						}
+
+						// Reserved Implicit Activation
+						for (const implicitActivationEventPrefix of reservedImplicitActivationEventPrefixes) {
+							if (activationEvent.startsWith(implicitActivationEventPrefix)) {
+								const start = document.positionAt(activationEventNode.offset);
+								const end = document.positionAt(activationEventNode.offset + activationEventNode.length);
+								diagnostics.push(new Diagnostic(new Range(start, end), implicitActivationEvent, DiagnosticSeverity.Error));
 							}
 						}
+
+						// Star activation
+						if (activationEvent === '*') {
+							const start = document.positionAt(activationEventNode.offset);
+							const end = document.positionAt(activationEventNode.offset + activationEventNode.length);
+							const diagnostic = new Diagnostic(new Range(start, end), starActivation, DiagnosticSeverity.Information);
+							diagnostic.code = {
+								value: 'star-activation',
```

Comment:
This is set in case we also offer code actions, do we want to offer a code action to remove it here?

Reply:
@joyceerhl that's set to show the link:
<img width="699" alt="image" src="https://user-images.githubusercontent.com/2644648/216439086-8cd675e2-a523-4832-a5bb-0f3bd53bc672.png">

which, when clicked, shows this: https://code.visualstudio.com/api/references/activation-events#Start-up